packages/stable/src/api/functions/functionsApi.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  BaseResourceAPI,
+  type CDFHttpClient,
+  type HttpResponse,
+  type MetadataMap,
+} from '@cognite/sdk-core';
+import { FunctionCallsApi } from './functionCallsApi';
+import type {
+  Function as CogniteFunction,
+  FunctionCallRequest,
+  FunctionIdEither,
+  FunctionListResponse,
+  FunctionListScope,
+  FunctionsActivationResponse,
+  FunctionsLimitsResponse,
+  IgnoreUnknownIdsField,
+  LimitList,
+} from './types.gen';
+
+export class FunctionsAPI extends BaseResourceAPI<CogniteFunction> {
+  private readonly functionCallsAPI: FunctionCallsApi;
+
+  constructor(...args: [string, CDFHttpClient, MetadataMap]) {
+    super(...args);
+
+    const [baseUrl, httpClient, map] = args;
+    this.functionCallsAPI = new FunctionCallsApi(baseUrl, httpClient, map);
+  }
+
+  /**
+   * @hidden
+   */
+  public get calls() {
+    return this.functionCallsAPI;
+  }
+
+  /**
+   * [List functions](https://api-docs.cognite.com/20230101/tag/Functions/operation/getFunctions)
+   *
+   * ```js
+   * client.functions.list({ limit: 100 })
+   * ```
+   *
+   */
+  async list(scope?: LimitList): Promise<HttpResponse<FunctionListResponse>> {
+    return await this.post<FunctionListResponse>(this.listPostUrl, {
+      data: scope || { limit: 100 }, // default limit
+    });
+  }
```

Comment:
```suggestion
  async list(scope?: LimitList): Promise<HttpResponse<FunctionListResponse>> {
    return super.listEndpoint(this.callListEndpointWithPost, scope);
  }
```
super.listEndpoint will do a bit more for free (e.g. auto pagination)

Reply:
The reason why is because the API does not support pagination